src/sync/sync_permission.cpp

```diff
@@ -61,6 +61,38 @@ Permission::AccessLevel extract_access_level(Object& permission, CppContext& con
 
 }
 
+// MARK: - Permission
+
+std::string Permission::description_for_access_level(AccessLevel level)
+{
+    switch (level) {
+        case AccessLevel::None: return "none";
+        case AccessLevel::Read: return "read";
+        case AccessLevel::Write: return "write";
+        case AccessLevel::Admin: return "admin";
+    }
+    REALM_UNREACHABLE();
+}
+
+bool Permission::paths_are_equivalent(std::string path_1, std::string path_2,
```

Comment:
I'm 99% sure we don't support tilde-substitution anywhere besides in a leading `/~/`. Your other concern is definitely valid. Not sure if we support `~` in a Realm name.